### PR TITLE
Nanosound - Infrared remote fix for Kernel 4.19

### PR DESCRIPTION
**Pull request by nanomesher**

### CODE REVIEW

**Comment by nanomesher** [comment]
This fix is essential to run with volumio 2.599 or above , which has kernel 4.19.  Kernel 4.19 has changed the implementation for LIRC

File: plugins/miscellanea/nanosound/install.sh [hunk 1]
```diff
@@ -5,22 +5,21 @@ sudo apt-get update
...
 
 #START OF python devs
-sudo apt-get -y install i2c-tools python-smbus python-pip python-dev python-imaging python-rpi.gpio python-lirc
-
-
-sudo pip install --upgrade setuptools
-sudo easy_install -U pip
-sudo pip install --upgrade python-mpd2 socketIO-client
+sudo apt-get -y install i2c-tools python-smbus python-pip python-dev python-imaging python-rpi.gpio
+sudo apt-get -y remove python-pip
```
Comment (volumio) [inline plugins/miscellanea/nanosound/install.sh]: Why this is necessary? This might compromise the system. Can this be avoided?

Comment (volumio) [reply plugins/miscellanea/nanosound/install.sh]: Isn't there any other solution? This might affect all packages working with python...

Comment (nanomesher) [reply plugins/miscellanea/nanosound/install.sh]: But the combined results of these 2 lines together below will just do an upgrade of pip, why would it affect?

sudo apt-get -y remove python-pip
 sudo -H easy_install pip

Comment (nanomesher) [reply plugins/miscellanea/nanosound/install.sh]: Is there any update to this? thanks

**Comment by balbuze** [comment]
Hi! Can you have a look for conflicting files please?
Why Kodi?

**Comment by nanomesher** [comment]
Not sure what happened, just did a diff and no changes has been made to plugins/miscellanea/kodi/UIconfig.json .

File: plugins/miscellanea/nanosound/install.sh [hunk 2]
```diff
@@ -5,22 +5,21 @@ sudo apt-get update
...
-echo "Creating lircrc file"
-sudo touch /etc/lirc/lircrc
+#sudo apt-get -y install lirc
+#echo "Applying LIRC starting policy"
+#systemctl disable lirc.service
+#systemctl stop lirc.service
+sudo apt-get -y remove lirc
+
+#echo "Creating lircrc file"
+#touch /etc/lirc/lircrc
```
Comment (arthurlutz) [inline plugins/miscellanea/nanosound/install.sh]: so LIRC is completely out of the picture ? Does this mean that if LIRC needs to be installed for some other usage it's not compatible ?

**Comment by arthurlutz** [comment]
Another general comment : I think the commented lines can be removed, git history can be used if necessary to look at the history.

**Comment by WarmChocolateCake** [comment]
As a general comment, I don't think an installation script should remove software (in this case pip).

This should be part of specific upgrade instructions for this release, as future updates would not require this extra step.

Also, it's possible that other software is installed with pip and this installer should not remove it.

**Comment by balbuze** [comment]
@nanomesher I'm a bit confused with this this PR. It is ready ?

**Comment by nanomesher** [comment]
sorry about the confusion. Taken feedback from comment above:
- avoided to affect existing LIRC installation in case other plugin needs it
- added a line to upgrade pip.  It  was our bad that it was missed.

Please merge and redeploy.

File: plugins/miscellanea/nanosound/install.sh (hunk 2 above)
Comment (nanomesher) [reply plugins/miscellanea/nanosound/install.sh]: we have avoided the need to remove LIRC.  Nanosound does not depend on it, but it's not nessessary to remove it.

**Comment by nanomesher** [comment]
> As a general comment, I don't think an installation script should remove software (in this case pip & lirc).
> 
> Edit: I see that pip is effectively being upgraded with a removal then install, however, my comment still stands for lirc - ie, would this installation script work on a clean install?
> 
> I think this should be part of specific upgrade instructions for this release, as future updates would not require these extra steps.

Points taken is the latest checkins about LIRC. thanks.
Regarding pip, there are some bugs in the pip upgrade so that it must be uninstalled and reinstalled to get around it.  The end result of the script is that pip will be upgraded and working nicely,

**Comment by balbuze** [comment]
Please, can you fix the conflicting file?

**Comment by nanomesher** [comment]
Conflict file has been fixed (plugins/miscellanea/kodi/UIConfig.json)
Please merge and then:
- Deploy nanosound_cd v1.1.5 to x86 and arm as the changes are in this PR
- Deploy nanosound v1.8.1 to arm only

thanks very much

**Comment by nanomesher** [comment]
@balbuze , are there any outstanding things we need to do before you can merge and deploy? thanks

**Comment by volumio** [comment]
@nanomesher there are still conflicts in the PR.
Regarding PIP please try to find a way where you don't need to uninstall and then reinstall. This can create problems when updating to future versions. Please let me know if you think there is an alternative way

**Comment by nanomesher** [comment]
Closing this one and will create 2 PR instead. One for NanoSound CD , One for NanoSound

**Pull request closed without merging**